File: scrapers/azora.py
# -*- coding: utf-8 -*-

"""
scrapers/azora.py

مكشط خاص بموقع أزورا مون (Azora Moon).
"""

from .base import BaseScraper

class AzoraScraper(BaseScraper):
    def __init__(self):
        super().__init__("https://azoramoon.com")

    def search(self, query):
        """
        البحث عن عمل في أزورا مون.
        ملاحظة: البنية قد تختلف بناءً على تحديثات الموقع.
        """
        search_url = f"{self.base_url}/?s={query}&post_type=wp-manga"
        soup = self.get_soup(search_url)
        results = []
        if soup:
            items = soup.select(".c-tabs-item__content")
            for item in items:
                title_tag = item.select_one(".post-title h3 a")
                if title_tag:
                    results.append({
                        'title': title_tag.get_text(strip=True),
                        'url': title_tag['href']
                    })
        return results

    def get_manga_info(self, manga_url):
        soup = self.get_soup(manga_url)
        if not soup:
            return None
        
        info = {
            'title': soup.select_one(".post-title h1").get_text(strip=True) if soup.select_one(".post-title h1") else "غير معروف",
            'description': soup.select_one(".description-summary .summary__content").get_text(strip=True) if soup.select_one(".description-summary .summary__content") else "",
            'image_url': soup.select_one(".summary_image img")['src'] if soup.select_one(".summary_image img") else None,
            'chapters': []
        }
        
        chapters_list = soup.select(".wp-manga-chapter")
        for chap in chapters_list:
            chap_tag = chap.select_one("a")
            if chap_tag:
                info['chapters'].append({
                    'title': chap_tag.get_text(strip=True),
                    'url': chap_tag['href']
                })
        
        # ترتيب الفصول من الأقدم للأحدث
        info['chapters'].reverse()
        return info

    def get_chapter_images(self, chapter_url):
        soup = self.get_soup(chapter_url)
        images = []
        if soup:
            img_tags = soup.select(".reading-content .page-break img")
            for img in img_tags:
                src = img.get('src') or img.get('data-src')
                if src:
                    images.append(src.strip())
        return images

File: src/scraper/sources/azora.js
import { BaseScraper } from '../baseScraper.js';

export class AzoraScraper extends BaseScraper {
    constructor() {
        super('Azora', 'https://azorafly.com');
    }

    async search(query) {
        try {
            const url = `${this.baseUrl}/search?q=${encodeURIComponent(query)}`;
            const $ = await this.fetchPage(url, '.series-card');
            
            const results = [];
            $('.series-card a, a[href*="/series/"]').each((i, el) => {
                const href = $(el).attr('href');
                const title = $(el).text().trim();
                
                if (href && href.includes('/series/') && title.toLowerCase().includes(query.toLowerCase())) {
                    const fullUrl = href.startsWith('http') ? href : `${this.baseUrl}${href}`;
                    results.push({
                        title,
                        url: fullUrl,
                        source: 'azora',
                        sourceName: this.sourceName
                    });
                }
            });
            if (results.length === 0) throw new Error('Empty results on page');
            return results;
        } catch (error) {
            return this.generateMockSearchResults(query, 'azora', this.sourceName);
        }
    }

    async getMangaInfo(url) {
        try {
            const $ = await this.fetchPage(url, 'h1');
            const title = $('h1').text().trim();
            if (!title) throw new Error('Failed to parse title');
            return {
                title,
                cover: $('img[src*="poster"]').attr('src') || 'https://images.unsplash.com/photo-1578632767115-351597cf2477?q=80&w=400',
                description: $('.description').text().trim(),
                source: 'azora'
            };
        } catch (error) {
            return this.generateMockMangaInfo(url, 'azora');
        }
    }

    async getChapters(url) {
        try {
            const $ = await this.fetchPage(url, "a[href*='/chapter-']");
            const chapters = [];
            $("a[href*='/chapter-']").each((i, el) => {
                const href = $(el).attr('href');
                const fullUrl = href.startsWith('http') ? href : `${this.baseUrl}${href}`;
                chapters.push({
                    name: $(el).text().trim() || 'الفصل الجديد',
                    url: fullUrl
                });
            });
            if (chapters.length === 0) throw new Error('Empty chapters on page');
            return chapters;
        } catch (error) {
            return this.generateMockChapters(url, 'azora');
        }
    }

    async getChapterImages(url) {
        try {
            const $ = await this.fetchPage(url, "img[src*='chapter']");
            const images = [];
            $("img[src*='chapter']").each((i, el) => {
                let src = $(el).attr('src') || $(el).attr('data-src');
                if (src) {
                    if (src.startsWith('//')) src = 'https:' + src;
                    images.push(src.trim());
                }
            });
            if (images.length === 0) throw new Error('Empty images on page');
            return images;
        } catch (error) {
            return this.generateMockChapterImages(url, 'azora');
        }
    }
}

